### PR TITLE
Clarify BeaconBlockAndBlobsSidecarByRoot no blob available

### PR DESCRIPTION
**Pull request by dapplion**
Peers may not know the slot of a block when requesting by root. So they may request a root via BeaconBlockAndBlobsSidecarByRoot which does not have any associated blob.
- Details of the edge case: https://github.com/ethereum/consensus-specs/issues/3113

This situation only applies before finalizing EIP4844_FORK_EPOCH. From discussions in https://github.com/ethereum/consensus-specs/pull/3139 a simple solution that requires a retry is the best trade-off to minimize complexity.

I picked the error code `3: ResourceUnavailable` without strong reasoning. Happy to change to another more meaningful one.

### CODE REVIEW

File: specs/eip4844/p2p-interface.md [hunk 1]
```diff
@@ -185,6 +185,9 @@ No more than `MAX_REQUEST_BLOCKS` may be requested at a time.
 The response MUST consist of zero or more `response_chunk`.
 Each _successful_ `response_chunk` MUST contain a single `SignedBeaconBlockAndBlobsSidecar` payload.
 
+If any root in the request content references a block outside of the range `[max(EIP4844_FORK_EPOCH, finalized_epoch, current_epoch - MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS), current_epoch]`
+ peers SHOULD respond with error code `3: ResourceUnavailable`.
+
 Clients MUST support requesting blocks and sidecars since the latest finalized epoch.
```
[the remark this answers is not in the document]
Comment (djrtwo) [reply specs/eip4844/p2p-interface.md:191]: Additionally, do we need to define a range? or just a minimum?

**Comment by dapplion** [comment]
@djrtwo thanks for the feedback, applied comments. Should there be a sentence noting the consequences of that line if `finalized_epoch < current_epoch - MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS`?
- Related to https://github.com/ethereum/consensus-specs/pull/3141

**Comment by djrtwo** [comment]
@dapplion Is that because you don't think the `max` is clear enough and want to make sure it's called out explictly? or something else

**Comment by dapplion** [comment]
Wording is clear to us, but the consequences as discussed today are important. So I feel that (for new readers for example) calling out that nuance is a useful pointer

**Review by Seyvick — APPROVED** [review]

**Review by Seyvick — APPROVED** [review]

**Review by djrtwo — APPROVED** [review]
I'm fine with an explicit line @dapplion but also happy to merge as is

**Review by g11tech — APPROVED** [review]
Lgtm

**Comment by djrtwo** [comment]
merging! feel free to add an additional comment in another PR

**Comment by Seyvick** [comment]
😱...estoy atónito!!!!

**Pull request merged**